cmd/podman/pod_rm.go

```diff
@@ -41,46 +40,18 @@ func podRmCmd(c *cli.Context) error {
 	if err := checkMutuallyExclusiveFlags(c); err != nil {
 		return err
 	}
+
 	runtime, err := libpodruntime.GetRuntime(c)
 	if err != nil {
 		return errors.Wrapf(err, "could not get runtime")
 	}
 	defer runtime.Shutdown(false)
 
-	args := c.Args()
 	ctx := getContext()
-	var delPods []*libpod.Pod
-	var lastError error
-	if c.Bool("all") {
-		delPods, err = runtime.GetAllPods()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get pod list")
-		}
-	}
-
-	if c.Bool("latest") {
-		delPod, err := runtime.GetLatestPod()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get latest pod")
-		}
-		delPods = append(delPods, delPod)
-	}
-
-	for _, i := range args {
-		pod, err := runtime.LookupPod(i)
-		if err != nil {
-			logrus.Errorf("%q", lastError)
-			if lastError != nil {
-				logrus.Errorf("%q", lastError)
-			}
-			lastError = errors.Wrapf(err, "unable to find pods %s", i)
-			continue
-		}
-		delPods = append(delPods, pod)
-	}
 	force := c.Bool("force")
+	pods, lastError := getPodsFromContext(c, runtime)
```

Comment:
I think we want to catch and handle errors from this separately from lastError

Reply:
It'll be more code, but a lot clearer as to what is actually going on

Reply:
I don't really agree with it, but you and @rhatdan wanted it so I'll go for it.